scrapy/commands/crawl.py

```diff
@@ -55,7 +55,10 @@ def run(self, args, opts):
         spname = args[0]
 
         self.crawler_process.crawl(spname, **opts.spargs)
+        crawler = list(self.crawler_process.crawlers)[0]
```

Comment:
What happens if an error occurs on the second crawler? Or is there never a second crawler?

Reply:
Aha. From the source I saw a comment that multiple spider is deprecated, so thought there's only one instantiation. Thanks for letting me know. So there are indeed multiple crawlers?

Reply:
```suggestion
        crawler = list(self.crawler_process.crawlers)
        self.crawler_process.start()
        closed_by_error_count = any([x.stats.get_value('finish_reason') == 'closespider_errorcount' for x in crawler])
```
Then we might try this instead?

Reply:
It looks better to me.

Although I believe `any` accepts an iterator, so you could use `any(…)` instead of `any([…])`.

Reply:
Well, I've used list comprehension to convert each crawler `x` to `x.stats.get_value('finish_reason') == 'closespider_errorcount'`, a boolean variable. I guess I can use map and lambda to do the same thing, like below.
```suggestion
        self.crawler_process.crawl(spname, **opts.spargs)
        self.crawler_process.start()

        closed_by_error_count = any(map(lambda x: x.stats.get_value('finish_reason') == 'closespider_errorcount',
                                        self.crawler_process.crawlers))
```